yash-semantics/src/command_impl/compound_command/case.rs

```diff
@@ -0,0 +1,403 @@
+// This file is part of yash, an extended POSIX shell.
+// Copyright (C) 2022 WATANABE Yuki
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+//! Execution of the case command
+
+use crate::expansion::attr::AttrChar;
+use crate::expansion::expand_word;
+use crate::expansion::expand_word_attr;
+use crate::Command;
+use crate::Handle;
+use std::ops::ControlFlow::Continue;
+use yash_env::semantics::ExitStatus;
+use yash_env::semantics::Result;
+use yash_env::Env;
+use yash_fnmatch::Config;
+use yash_fnmatch::Pattern;
+use yash_fnmatch::PatternChar;
+use yash_syntax::syntax::CaseItem;
+use yash_syntax::syntax::Word;
+
+/// Converts unquoted backslashes to quoting characters.
+///
+/// Sets the `is_quoting` flag of unquoted backslashes and the `is_quoted` flag
+/// of their following characters.
+fn apply_escapes(chars: &mut [AttrChar]) {
+    if !chars.is_empty() {
+        for i in 0..chars.len() - 1 {
```

Comment:
Remove the `is_empty` check.

```suggestion
    for j in 1..chars.len() {
        let i = j - 1;
```